src/file_watcher.rs

```diff
@@ -0,0 +1,1417 @@
+            .header("Content-Type", "text/plain; charset=utf-8")
+            .header("Accept", "application/json")
+            .header(
+                "Authorization",
+                format!("Token {}", config()?.influxdb_token.clone()),
+            )
+            .build()?)
+    }
+
+    fn enforce_singleton(&self, lockfile_path: &Path) -> Result<Flock<File>> {
+        enforce_singleton(lockfile_path)
+    }
+
+    fn poll_directory(&self, file_watcher: &FileWatcher) -> Result<bool> {
+        file_watcher.poll_directory()
+    }
+
+    fn exit(&self, code: i32) {
+        exit(code)
+    }
+
+    fn sleep(&self, duration: Duration) {
+        sleep(duration)
+    }
+}
+
+struct DefaultStartHelpers;
+impl StartHelpers for DefaultStartHelpers {}
+
+trait PollDirectoryHelpers {
+    fn find_telemetry_files(&mut self, ignore_age: bool) -> Result<Vec<PathBuf>> {
+        find_telemetry_files(ignore_age)
+    }
+
+    fn flock(
+        &self,
+        file: File,
+        flags: FlockArg,
+    ) -> std::result::Result<Flock<File>, (File, nix::errno::Errno)> {
+        Flock::lock(file, flags)
+    }
+
+    fn buffered_reader<R: Read>(
+        &self,
+        file: R,
+    ) -> std::result::Result<Vec<String>, std::io::Error> {
+        BufReader::new(file)
+            .lines()
+            .collect::<std::result::Result<Vec<_>, _>>()
+    }
+
+    fn base64_decode(&self, line: &str) -> std::result::Result<Vec<u8>, DecodeError> {
+        STANDARD.decode(line.as_bytes())
+    }
+
+    fn string_from_utf8(
+        &mut self,
+        bytes: Vec<u8>,
+    ) -> std::result::Result<String, std::string::FromUtf8Error> {
+        String::from_utf8(bytes)
+    }
+
+    #[allow(clippy::type_complexity)]
+    fn parse_event<'a>(
+        &self,
+        line: &'a str,
+    ) -> Result<(
+        Captures<'a>,
+        Vec<String>,
+        Vec<String>,
+        HashMap<&'a str, String>,
+    )> {
+        let event = TELEMETRY_EVENT_REGEX
+            .as_ref()?
+            .captures(line)
+            .ok_or_else(|| TelemetryError::InvalidTracingEvent(line.to_string()))?;
+
+        let payload = event
+            .name("payload")
+            .ok_or_else(|| TelemetryError::InvalidTracingPayload(line.to_string()))?;
+
+        let (mut spans_long, mut spans_short, mut fields) = (vec![], vec![], HashMap::new());
+
+        for capture in TELEMETRY_PAYLOAD_REGEX
+            .as_ref()?
+            .captures_iter(payload.into())
+        {
+            if let Some(span) = capture.name("span") {
+                let span_str = span.as_str();
+                spans_long.push(span_str.to_string());
+                spans_short.push(span_str.split('{').next().unwrap_or(span_str).to_string());
+            } else if let Some(message) = capture.name("message") {
+                fields.insert("message", message.as_str().to_string());
+            } else if let Some(message_only) = capture.name("message_only") {
```

Comment:
nit: magic strings; could be constants

Reply:
I just tried moving these out, but noticed they're not really magic strings - they're related to regex captures from `TELEMETRY_EVENT_REGEX` and `TELEMETRY_PAYLOAD_REGEX`. So if these were to be turned into constants, it would make the regular expression look unnatural especially given that `{` and `}` themselves mean special things within a regular expression.